apps/freeablo/faworld/item.h

```diff
@@ -19,121 +19,116 @@ namespace FARender
+        int32_t getMinAttackDamage() const;
+        int32_t getMaxAttackDamage() const;
 
-        bool operator==(const Item& rhs) const { return this->mUniqueId == rhs.mUniqueId; }
+    private:
+        std::string chargesStr() const;
+        std::string damageStr() const;
+        std::string armorStr() const;
+        std::string damageOrArmorStr() const;
+        std::string durabilityStr() const;
+        std::string requirementsStr() const;
+        const DiabloExe::BaseItem& base() const;
 
-        ItemType getType() const { return mType; }
-        ItemClass getClass() const { return mClass; }
-        ItemEquipType getEquipLoc() const { return mEquipLocation; }
+    private:
+        static Cel::CelFile* mObjcurs;
+        DiabloExe::Affix mAffix;
 
-        uint32_t getGraphicValue() const { return mGraphicValue; }
-        const std::string& getName() const { return mName; }
-        const std::string& getFlipAnimationPath() const { return mDropItemGraphicsPath; }
-        bool isEmpty() const { return mEmpty; }
-        std::pair<uint8_t, uint8_t> getInvSize() const { return std::pair<uint8_t, uint8_t>(mSizeX, mSizeY); }
-        std::pair<uint8_t, uint8_t> getInvCoords() const { return std::pair<uint8_t, uint8_t>(mInvX, mInvY); }
-        std::pair<uint8_t, uint8_t> getCornerCoords() const { return std::pair<uint8_t, uint8_t>(mCornerX, mCornerY); }
-        bool isBeltEquippable() const { return mSizeX == 1 && mSizeY == 1 && mUseOnce && mType != ItemType::gold; }
+        ItemId mBaseId;
+        int32_t mMaxDurability;
+        int32_t mCurrentDurability;
+        int32_t mArmorClass;
 
-        // public members
-        bool mIsReal = false;
-        uint32_t mUniqueId = 0;
-        uint32_t mCount = 0;
+        uint8_t mCornerX;
+        uint8_t mCornerY;
 
-        static Item empty;
+        bool mEmpty = true;
+        bool mIsIdentified;
+
+        int32_t mUniqueId = 0;
 
     private:
-        static std::unique_ptr<Cel::CelFile> mObjcurs;
-
-        std::string mName;
-        ItemType mType = ItemType::misc;
-        ItemClass mClass = ItemClass::none;
-        ItemEquipType mEquipLocation = ItemEquipType::none;
-        uint32_t mGraphicValue = 0;
-        bool mUseOnce = 0;
-        std::string mDropItemGraphicsPath;
-        uint32_t mMaxCount = 0;
-        uint8_t mBaseId = 0;
-        bool mEmpty = true;
+        int32_t mMaxCharges = 0;
+        int32_t mCurrentCharges = 0;
 
         // TODO: these should be handled by inventory class, not item class
         uint8_t mInvY = 0;
         uint8_t mInvX = 0;
 
-        uint8_t mSizeX = 0;
-        uint8_t mSizeY = 0;
-
-        uint8_t mCornerX = 0;
-        uint8_t mCornerY = 0;
+        const DiabloExe::DiabloExe* mExe = nullptr; // since we want to restore base info from exe data
 
         friend class Inventory;
-
-        // variables that used to be in this class, loaded form the exe, but ommented out for now sine we don't use them.
```

Comment:
I wouldn't get rid of these

Reply:
Well I removed them because I replaced most of them with functions which return stuff from `DiabloExe` for now since without affixes almost anything is constant anyway. With affixes some sort of state would be required but I think it'd be better to separate it from `Item` itself and store it as some other class inside it because for example it does not need to be serialized. But anyway I think it's something which needs to be thought through a bit later.

Reply:
Ok, that makes sense

Reply:
It's a matter of choice really, whether to read original files on the fly or convert at first launch. I opeted for read on the fly, and don't plan to change that for now unless there's a compelling reason down the line.

Reply:
Modification of resources for modders.

Reply:
Sure, but mods aren't a priority until the game works. And besides, we can already load pngs and the likes anyway, we don't need to convert all the cels to pngs for that